nucypher/acumen/perception.py

```diff
@@ -358,8 +378,65 @@ def record_remote_fleet_state(self,
                                   state_checksum: str,
                                   timestamp: maya.MayaDT,
                                   population: int):
+
+        if checksum_address not in self._current_state:
+            raise KeyError(f"A node {checksum_address} is not present in the current fleet state")
+
         nickname = Nickname.from_seed(state_checksum, length=1)
-        self.remote_states[checksum_address] = ArchivedFleetState(checksum=state_checksum,
-                                                                  nickname=nickname,
-                                                                  timestamp=timestamp,
-                                                                  population=population)
+        state = ArchivedFleetState(checksum=state_checksum,
+                                   nickname=nickname,
+                                   timestamp=timestamp,
+                                   population=population)
+
+        self._remote_last_seen[checksum_address] = maya.now()
+        self._remote_states[checksum_address] = state
+
+    def status_info(self, checksum_address_or_node: Union[ChecksumAddress, 'Ursula']) -> 'RemoteUrsulaStatus':
+
+        if isinstance(checksum_address_or_node, str):
+            node = self[checksum_address_or_node]
```

Comment:
Do you think it is necessary to handle a `KeyError` here in case of invalid or unexpected input?